azure-kusto-data/azure/kusto/data/client.py

```diff
@@ -853,19 +853,61 @@ def execute_streaming_ingest(
 
         self._execute(endpoint, database, None, stream, self._streaming_ingest_default_timeout, properties)
 
-    def _execute(
-        self, endpoint: str, database: str, query: Optional[str], payload: Optional[io.IOBase], timeout: timedelta, properties: ClientRequestProperties = None
+    def execute_streaming_query(
+        self, database: str, query: str, timeout: timedelta = _KustoClientBase._query_default_timeout, properties: Optional[ClientRequestProperties] = None
+    ) -> Response:
+        """
+        Query directly from Kusto database using streaming output.</p>
+        This method queries the Kusto database into an asyncio stream.
+
+        example usage::
+
+            my_file = open("some_file", "wb")
+            with client.execute_streaming_query(...) as response:
+                for chunk in response.iter_content(1024):
+                    file.write(chunk)
+
+        :return: a Response that can be read as chunks
+        """
+        response = self._retrieve_response(self._query_endpoint, database, properties, query, None, timeout, stream=True)
+
+        try:
+            response.raise_for_status()
+        except Exception as e:
+            self._handle_http_error(e, self._query_endpoint, None, response, response.json(), response.text)
+
+        return response
+
+    def _retrieve_response(
+        self,
+        endpoint: str,
+        database: str,
+        properties: Optional[ClientRequestProperties],
+        query: str,
+        payload: Optional[io.IOBase],
+        timeout: timedelta,
+        stream: bool,
     ):
-        """Executes given query against this client"""
         request_params = ExecuteRequestParams(database, payload, properties, query, timeout, self._request_headers)
         json_payload = request_params.json_payload
         request_headers = request_params.request_headers
         timeout = request_params.timeout
-
         if self._auth_provider:
             request_headers["Authorization"] = self._auth_provider.acquire_authorization_header()
+        response = self._session.post(endpoint, headers=request_headers, json=json_payload, data=payload, timeout=timeout.seconds, stream=stream)
+        return response
 
-        response = self._session.post(endpoint, headers=request_headers, data=payload, json=json_payload, timeout=timeout.seconds)
+    def _execute(
+        self,
+        endpoint: str,
+        database: str,
+        query: Optional[str],
+        payload: Optional[io.IOBase],
+        timeout: timedelta,

```

Comment:
I believe timeout is also an optional

Reply:
See before

Reply:
Sorry -  
Not yours: I believe timeout is also an optional